src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlServiceImpl.java

```diff
@@ -0,0 +1,161 @@
+        Supplier supplier = new Supplier();
+        supplier.setActive(JaggaerBuilder.buildActive(JaggaerConstants.YES));
+        supplier.setErpNumber(JaggaerBuilder.buildERPNumber(erpNumber));
+        supplier.setName(JaggaerBuilder.buildName(name));
+        supplier.setAddressList(new AddressList());
+        supplier.getAddressList().getAddress().add(buildTestAddress(addressLine1));
+        supplier.getAddressList().getAddress().add(buildTestAddress(addressLine2));
+        return supplier;
+    }
+
+    private Address buildTestAddress(String addressLine) {
+        Address address = new Address();
+        address.setActive(JaggaerBuilder.buildActive(JaggaerConstants.YES));
+        address.setAddressLine1(JaggaerBuilder.buildAddressLine(addressLine));
+        return address;
+    }
+
+    protected List<SupplierSyncMessage> buildSupplierSyncMessageList(List<Supplier> suppliers,
+            int maximumNumberOfSuppliersPerListItem) {
+        List<SupplierSyncMessage> messages = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(suppliers)) {
+            for (List<Supplier> supplierChunk : ListUtils.partition(suppliers, maximumNumberOfSuppliersPerListItem)) {
+                SupplierSyncMessage message = new SupplierSyncMessage();
+                message.setVersion(JaggaerConstants.XML_VERSION);
+                message.setHeader(buildHeader());
+                SupplierRequestMessage supplierRequest = new SupplierRequestMessage();
+                supplierRequest.getSupplier().addAll(supplierChunk);
+
+                message.getSupplierRequestMessageItems().add(supplierRequest);
+                messages.add(message);
+            }
+        }
+        LOG.info("buildSupplierSyncMessageList, found {} suppliers, and returning {} SupplierSyncMessages", suppliers.size(), messages.size());
+        return messages;
+    }
+
+    private Header buildHeader() {
+        Header header = new Header();
+        header.setMessageId(UUID.randomUUID().toString());
+        header.setTimestamp(DATE_FORMATTER.print(dateTimeService.getCurrentDate().getTime()));
+        header.setAuthentication(buildAuthentication());
+        return header;
+    }
+
+    /*
+     * @todo implement this with real values after we get credential details from
+     * Jaggaer on KFSPTS-28268 if not sooner in the project
+     */
+    private Authentication buildAuthentication() {
+        Authentication auth = new Authentication();
+        auth.setIdentity("CU - Identity");
+        auth.setSharedSecret("CU - Share Secret");
+        return auth;
+    }
+
+    @Override
+    public void generateXMLForSyncMessages(List<SupplierSyncMessage> messages) {
+        for (SupplierSyncMessage message : messages) {
+            String outputFileName = jaggaerXmlDirectory + JAGGAER_UPLOAD_FILE_NAME
+                    + DATE_FORMATTER.print(dateTimeService.getCurrentDate().getTime())
+                    + CUKFSConstants.XML_FILE_EXTENSION;
```

Comment:
Since this code is potentially writing multiple new files in rapid succession, are there any concerns about earlier files accidentally getting overwritten by later files (given that the timestamp is the only unique portion), or is the generation slow enough (and the timestamp specific enough) to prevent that?

Reply:
Ok, sounds good. Thanks for clarifying!

Reply:
Jay, are there really slashes '/' embedded in the file name, or are those path separators?

Reply:
@dp65 oh,  let me take a look at that ... I used a library to build that out, let me see if I can adjust that

Reply:
After reviewing this again based on Donivan's comment, I noticed one other date formatting change that might be needed. I'll add a comment at the related spot.